modules/gdscript/gdscript_functions.cpp

```diff
@@ -729,22 +729,16 @@ void GDScriptFunctions::call(Function p_func, const Variant **p_args, int p_arg_
 				r_ret = Variant();
 				return;
 			}
+			r_ret = *p_args[0];
 
 			VariantParser::StreamString ss;
 			ss.s = *p_args[0];
 
 			String errs;
 			int line;
-			Error err = VariantParser::parse(&ss, r_ret, errs, line);
-
-			if (err != OK) {
-				r_error.error = Variant::CallError::CALL_ERROR_INVALID_ARGUMENT;
-				r_error.argument = 0;
-				r_error.expected = Variant::STRING;
-				r_ret = "Parse error at line " + itos(line) + ": " + errs;
-				return;
-			}
+			Error parse_error = VariantParser::parse(&ss, r_ret, errs, line);
 
+			ERR_FAIL_COND(parse_error != OK)
```

Comment:
Hmm... Good point. The message would probably be quite cryptic and indeed annoying for those who are handling it in script. So do you suggest that it's best to not notify about the error at all?

Reply:
IMO it's best to have no error message. I don't think there's a way to make the message clear and be able to dismiss it when needed.

But this behavior should also be documented.

Reply:
Well, I could see a way of making it clear; you could do something like this:

```cpp
const bool str2var_input_string_parse_error= parse_error != OK;

ERR_FAIL_COND(str2var_input_string_parse_error);
```
Which would produce error message `Condition 'str2var_input_string_parse_error' is true.` and this looks pretty clear to me. The code is a bit more verbose, but it does the job.

The fact still remains, though, that the error message couldn't be disabled, which would be really annoying in case when the user is aware of the parse error. So in the end, I guess I agree that the best would be not to print anything.

Also, documenting this behaviour is indeed a must, it's not quite what one might expect.

Reply:
If you want a proper error message you can use the `ERR_EXPLAIN` macro before the `ERR_FAIL_COND`. I don't think it's good without being able to dismiss it though, so I prefer to just remove it.

Reply:
Removed `ERR_FAIL_COND`. Marking conversation as resolved.